File: commcare-core/src/test/java/org/javarosa/xpath/expr/test/XPathJsonPropertyFuncTest.java
package org.javarosa.xpath.expr.test;

import org.javarosa.xpath.expr.XPathJsonPropertyFunc;
import org.json.JSONException;
import org.junit.Assert;
import org.junit.Test;

/**Tests for the XPathJsonProperty
 *
 * @author rcostello
 */

public class XPathJsonPropertyFuncTest {

    @Test
    public void getJsonProperty() {
        String testObj1 = "{\"name\":\"Sam\"}";
        String testVal1 = XPathJsonPropertyFunc.Companion.getJsonProperty(testObj1, "name");
        String testVal2 = XPathJsonPropertyFunc.Companion.getJsonProperty(testObj1, "city");
        Assert.assertEquals(testVal1, "Sam");
        Assert.assertEquals(testVal2, "");

        String testObj2 = "{city: New York}";
        String testVal3 = XPathJsonPropertyFunc.Companion.getJsonProperty(testObj2, "city");
        String testVal4 = XPathJsonPropertyFunc.Companion.getJsonProperty(testObj2, "state");
        Assert.assertEquals(testVal3, "New York");
        Assert.assertEquals(testVal4, "");

        String testInvalidObj = "{\"name\"}: \"Sam\"}";
        String testVal5 = XPathJsonPropertyFunc.Companion.getJsonProperty(testInvalidObj, "name");  
        Assert.assertEquals(testVal5, "");
        
        String testEmptyStrObj = "";
        String testVal6 = XPathJsonPropertyFunc.Companion.getJsonProperty(testEmptyStrObj, "name");  
        Assert.assertEquals(testVal6, "");
    }
}

File: commcare-core/src/test/java/org/javarosa/xpath/expr/test/XPathJsonPropertyFuncTest.kt
package org.javarosa.xpath.expr.test

import org.javarosa.xpath.expr.XPathJsonPropertyFunc
import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Tests for the XPathJsonProperty
 *
 * @author rcostello
 */
class XPathJsonPropertyFuncTest {

    @Test
    fun getJsonProperty() {
        val testObj1 = """{"name":"Sam"}"""
        val testVal1 = XPathJsonPropertyFunc.getJsonProperty(testObj1, "name")
        val testVal2 = XPathJsonPropertyFunc.getJsonProperty(testObj1, "city")
        assertEquals(testVal1, "Sam")
        assertEquals(testVal2, "")

        val testObj2 = "{city: New York}"
        val testVal3 = XPathJsonPropertyFunc.getJsonProperty(testObj2, "city")
        val testVal4 = XPathJsonPropertyFunc.getJsonProperty(testObj2, "state")
        assertEquals(testVal3, "New York")
        assertEquals(testVal4, "")

        val testInvalidObj = """{"name"}: "Sam"}"""
        val testVal5 = XPathJsonPropertyFunc.getJsonProperty(testInvalidObj, "name")
        assertEquals(testVal5, "")

        val testEmptyStrObj = ""
        val testVal6 = XPathJsonPropertyFunc.getJsonProperty(testEmptyStrObj, "name")
        assertEquals(testVal6, "")
    }
}
